addrmgr/knownaddress.go

```diff
@@ -63,11 +63,13 @@ func (ka *KnownAddress) LastAttempt() time.Time {
 func (ka *KnownAddress) chance() float64 {
 	ka.mtx.Lock()
 	defer ka.mtx.Unlock()
-	now := time.Now()
-	lastAttempt := now.Sub(ka.lastattempt)
 
-	if lastAttempt < 0 {
-		lastAttempt = 0
+	var lastAttempt time.Duration
```

Comment:
Nice find!  I think this entire func would probably be simpler and easier to follow by just short circuiting though don't you think?

```Go
func (ka *KnownAddress) chance() float64 {
	defer ka.mtx.Unlock()
	ka.mtx.Lock()

	// Very recent attempts are less likely to be retried.
	const minChance = 0.01
	if ka.lastattempt.IsZero() || time.Since(ka.lastattempt) < 10*time.Minute {
		return minChance
	}

	// Failed attempts deprioritise.
	c := 1.0 / math.Pow(1.5, float64(ka.attempts))
	return math.Max(c, minChance)
}
```